src/Lecture2.hs

```
@@ -261,7 +326,13 @@ data EvalError
 It returns either a successful evaluation result or an error.
 -}
 eval :: Variables -> Expr -> Either EvalError Int
-eval = error "TODO"
+eval var (Lit n) = Right n
+eval var (Var str) = Right (fromMaybe 0 $lookup str var)
```

Your implementation never returns `Left` with some `EvalError`. By the task, this function should return an error when a variable not found.